server/config.go

```diff
@@ -235,6 +235,30 @@ func RunSetZone(ctx *Context, keyPrefix, configFileName string) {
 	runSetConfig(ctx, zonePathPrefix, keyPrefix, configFileName)
 }
 
+// putConfig writes a config for the specified key prefix (which is
+// treated as a key). The config is parsed from the input "body". The
+// config is stored gob-encoded. The specified body must validly
+// parse into a config struct and must pass a given validation check.
+func putConfig(db *client.KV, configPrefix proto.Key, config gogoproto.Message,
+	path string, body []byte, r *http.Request,
+	validate func(gogoproto.Message) error) error {
+	if len(path) == 0 {
+		return util.Errorf("no path specified for Put")
+	}
+	if err := util.UnmarshalRequest(r, body, config, util.AllEncodings); err != nil {
+		return util.Errorf("config has invalid format: %+v: %s", config, err)
+	}
+	if err := validate(config); err != nil {
```

Comment:
I would have allowed `validate` to be nil and only invoke it when non-nil. This would slightly simplify some of the invocations of `putConfig`.